### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by jaswanth115**
Javascript Home task problems

### CODE REVIEW

**Comment by jaswanth115** [comment]
Please review.

File: js2.1.4.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+function rentalCarCost(d) {
```
Comment (NirmalaEpam) [inline js2.1.4.js:1]: Give a meaningful parameter name and use the same in functions

File: js2.2.1.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+function expression(number, operation){
+	if(!operation)
+		return number;
+	return operation(number);
+}
+
+function zero(operation) { return expression(0, operation); }
+function one(operation) { return expression(1, operation); }
+function two(operation) { return expression(2, operation); }
+function three(operation) { return expression(3, operation); }
```
Comment (NirmalaEpam) [inline js2.2.1.js:10]: where are you giving the operation name? like expression(4,'plus') .. is this code returns the correct value?

File: js2.2.2.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+function getMiddle(s) {
+    if (s.length % 2 !== 0) {
+        var mid = Math.floor(s.length / 2);
```
Comment (NirmalaEpam) [inline js2.2.2.js:3]: follow naming convention

File: js2.2.2.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+function getMiddle(s) {
+    if (s.length % 2 !== 0) {
+        var mid = Math.floor(s.length / 2);
+        return s.slice(mid, mid + 1);
+    }
+    else {
+        var mid = s.length / 2;
+        return s.slice(mid - 1, mid + 1);
```
Comment (NirmalaEpam) [inline js2.2.2.js:8]: I see only the function definition , where is the function invocation??

File: js2.2.3.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+// partition the items array so that all values for which pred returns true are
+// at the end, returning the index of the first true value
+function partitionOn(pred, items) {
+    var sum_par = 0,
```
Comment (NirmalaEpam) [inline js2.2.3.js:4]: include function invoke  and include console logs to see the results .

File: js2.3.1.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+function prefill(n, v) {
+    if(n === 0 || n === '0') return [];
+   
+     if(!parseInt(n) || n < 0 ){
+   
+       let typeError = new TypeError();
```
Comment (NirmalaEpam) [inline js2.3.1.js:6]: provide meaningful name for parameters

File: js2.3.2.js [hunk 7]
```diff
@@ -0,0 +1,5 @@
+// return a function that returns n
+function always (n) {
+    return function () { return n };
```
Comment (NirmalaEpam) [inline js2.3.2.js:3]: naming conventinon

File: js2.3.3.js [hunk 8]
```diff
@@ -0,0 +1,11 @@
+function createFunctions(n) {
+    var callbacks = [];
```
Comment (NirmalaEpam) [inline js2.3.3.js:2]: provide description about the method and what it does

File: js2.3.3.js [hunk 9]
```diff
@@ -0,0 +1,11 @@
+function createFunctions(n) {
+    var callbacks = [];
+  
+    for (let i=0; i<n; i++) {
+      callbacks.push(function() {
```
Comment (NirmalaEpam) [inline js2.3.3.js:5]: follow naming conventions

File: js2.4.2.js [hunk 10]
```diff
@@ -0,0 +1,6 @@
+function add(n){
+    // Let the currying begin!
+    const f = x => add(n + x);
```
Comment (NirmalaEpam) [inline js2.4.2.js:3]: add description and follow naming conventions

File: js2.4.4.js [hunk 11]
```diff
@@ -0,0 +1 @@
+const compose = (f, g) => (...args) => f(g(...args));
```
Comment (NirmalaEpam) [inline js2.4.4.js:1]: add function invoke and console logs to explain about compose . follow naming conventions.. 
I want examples which uses this function

File: js2.5.1.js [hunk 12]
```diff
@@ -0,0 +1,18 @@
+Array.prototype.square = function() {
+    return this.map(v => v * v);
```
Comment (NirmalaEpam) [inline js2.5.1.js:2]: change all parameters name which will be related to function and examples has to be included

File: js2.5.2.js [hunk 13]
```diff
@@ -0,0 +1,9 @@
+// return the nested property value if it exists,
+// otherwise return undefined
+Object.prototype.hash = function(string) {
+    try {
```
Comment (NirmalaEpam) [inline js2.5.2.js:4]: include examples

File: js2.5.4.js [hunk 14]
```diff
@@ -0,0 +1,6 @@
+function isSantaClausable(obj) {
+    // TODO
+    return ['sayHoHoHo', 'distributeGifts', 'goDownTheChimney'].every(function(methodName) {
+      return typeof obj[methodName] == 'function';
```
Comment (NirmalaEpam) [inline js2.5.4.js:4]: include description and examples

**Pull request merged**